### PR TITLE
Bind context functions

### PR DESCRIPTION
**Pull request by zhirafovod**
## Description

Bind context function to TemplateProcessor instance, so they can access template processor methods, like setData. 

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: src/TemplateProcessor.ts [hunk 1]
```diff
@@ -179,11 +181,11 @@ export default class TemplateProcessor {
         this.changeCallbacks = new Map();
     }
 
-    private setupContext(context: {}) {
+    private setupContext() {
```
Comment (geoffhendrey) [inline src/TemplateProcessor.ts:184]: please add TS typings (parameters, return type)

**Review by geoffhendrey — CHANGES_REQUESTED** [review]

File: src/TemplateProcessor.ts (hunk 1 above)
Comment (zhirafovod) [reply src/TemplateProcessor.ts]: added return type (void). Should I somehow indicate that the function does not take any arguments?

File: src/TemplateProcessor.ts [hunk 2]
```diff
@@ -179,7 +181,7 @@ export default class TemplateProcessor {
         this.changeCallbacks = new Map();
     }
 
-    private setupContext(context: {}) {
+    private setupContext(context: {}): void {
         this.context = merge(
```
Comment (geoffhendrey) [inline src/TemplateProcessor.ts:185]: return merged value. Don't assign to this.context!

**Review by geoffhendrey — CHANGES_REQUESTED** [review]

**Comment by zhirafovod** [comment]
After discussions, this approach was found flawed and replaced with https://github.com/cisco-open/stated/pull/24

**Pull request closed without merging**